File: VB/ButtonEditWithPadding/RepositoryItemPaddingButtonEdit.vb
Imports Microsoft.VisualBasic
Imports System.Windows.Forms
Imports DevExpress.XtraEditors.Registrator
Imports DevExpress.XtraEditors.Repository

Namespace ButtonEditWithPadding
	<UserRepositoryItem("Register")> _
	Public Class RepositoryItemPaddingButtonEdit
		Inherits RepositoryItemButtonEdit
		Friend Const EditorName As String = "PaddingButtonEdit"
		Private _padding As Padding

		Shared Sub New()
			Register()
		End Sub
		Public Sub New()
		End Sub

		Public Shared Sub Register()
            EditorRegistrationInfo.Default.Editors.Add(New EditorClassInfo(EditorName, GetType(PaddingButtonEdit), GetType(RepositoryItemPaddingButtonEdit), GetType(PaddingButtonEditViewInfo), New PaddingButtonEditPainter(), True))
		End Sub

		Public Overrides ReadOnly Property EditorTypeName() As String
			Get
				Return EditorName
			End Get
		End Property
		Public Property Padding() As Padding
			Get
				Return _padding
			End Get
			Set(ByVal value As Padding)
				_padding = value
				Me.OnPropertiesChanged()
			End Set
		End Property
	End Class
End Namespace

File: CS/ButtonEditWithPadding/RepositoryItemPaddingButtonEdit.cs
using System.Windows.Forms;
using DevExpress.XtraEditors.Registrator;
using DevExpress.XtraEditors.Repository;

namespace ButtonEditWithPadding
{
    [UserRepositoryItem("Register")]
    public class RepositoryItemPaddingButtonEdit : RepositoryItemButtonEdit
    {
        internal const string EditorName = "PaddingButtonEdit";
        private Padding _padding;

        static RepositoryItemPaddingButtonEdit()
        {
            Register();
        }
        public RepositoryItemPaddingButtonEdit()
        {
        }

        public static void Register()
        {
            EditorRegistrationInfo.Default.Editors.Add(new EditorClassInfo(EditorName, typeof(PaddingButtonEdit),
                typeof(RepositoryItemPaddingButtonEdit), typeof(PaddingButtonEditViewInfo),
                    new PaddingButtonEditPainter(), true, null));
        }

        public override string EditorTypeName
        {
            get { return EditorName; }
        }
        public Padding Padding
        {
            get { return _padding; }
            set
            {
                _padding = value;
                this.OnPropertiesChanged();
            }
        }
    }
}